microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -192,46 +173,47 @@ A typical JavaScript object can be one of two array types:
 
 When there are a small number of properties, the properties are stored internally in the JavaScript object.
 
-**Map** is an object that describes both the kind of object it is and the layout.  For example, maps are used to describe implicit object hierarchies for [fast property access](https://v8.dev/blog/fast-properties).
-
-### Object groups
+**Map** is an object that describes both the kind of object it is and the layout. For example, maps are used to describe implicit object hierarchies for [fast property access](https://v8.dev/blog/fast-properties).
 
-Each _native objects group_ is made up of objects that hold mutual references to each other.  Consider, for example, a DOM subtree where every node has a link to the relative parent and links to the next child and next sibling, thus forming a connected graph.
+<!-- ------------------------------ -->
+#### Object groups
+<!-- This doesn't seem very actionable to users. We describe groups, but don't say where they appear in the tool and what's interesting about them. We then talk about native objects, seemingly out of the blue. -->
+Each _native object group_ is made up of objects that hold mutual references to each other. Consider, for example, a DOM subtree where every node has a link to the relative parent and links to the next child and next sibling, thus forming a connected graph.
 
-Note that native objects aren't represented in the JavaScript heap.  The lack of representation is why native objects have zero size.  Instead, wrapper objects are created.
+Note that native objects aren't represented in the JavaScript heap. The lack of representation is why native objects have zero size. Instead, wrapper objects are created.
 
-Each wrapper object holds a reference to the corresponding native object, for redirecting commands to it.  In turn, an object group holds wrapper objects.  This doesn't create an uncollectable cycle, because garbage collection is smart enough to release object groups whose wrappers are no longer referenced.  But forgetting to release a single wrapper will hold references to the whole group and to any associated wrappers.
+Each wrapper object holds a reference to the corresponding native object, for redirecting commands to it. In turn, an object group holds wrapper objects. This doesn't create an uncollectable cycle, because garbage collection is smart enough to release object groups whose wrappers are no longer referenced. But forgetting to release a single wrapper will hold references to the whole group and to any associated wrappers.
 
 
 <!-- ====================================================================== -->
 ## Cycles
-
+<!-- This needs a concrete example, not sure what earlier and later mean. And why is removing the earlier instance important? -->
```

Comment:
@sethbrenith any help on this would be great!

Reply:
Hmm, I agree that this section is pretty confusing, and I'm not quite sure how to make it better. The simplest advice would be: when trying to remove an object from the heap, it's generally most helpful to focus on the shortest retaining path, which is expanded by default when you select the object.

The Retainers pane shows all objects which point to the selected object. Following along edges backwards through the graph, you might end up revisiting a node you've already seen, which is not super useful, so Jose added a filter to hide such nodes in the Retainers view. However, I'm not sure that this section fits conceptually with the rest of this article: the article seems to be high-level ideas to get someone started on memory analysis, and this section is a description of a specific feature to solve a specific problem. It might be best to just remove this section.

Reply:
Thanks. I'll consider moving these notes to our Memory tool reference article instead then.